File: examples/c++/tsl2561.cxx
/*
 * Author: Nandkishor Sonar <Nandkishor.Sonar@intel.com>
 * Copyright (c) 2014 Intel Corporation.
 *
 * LIGHT-TO-DIGITAL CONVERTER [TAOS-TSL2561]
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <stdio.h>
#include <stdlib.h>

#include "tsl2561.hpp"

int
main(int argc, char** argv)
{
    int loopCount = 100;

    //! [Interesting]
    if (argc < 2) {
        printf("Provide loop count \n");
    } else {
        loopCount = atoi(argv[1]);
    }
    upm::TSL2561 sensor;

    for (int i = 0; i < loopCount; i++) {
        fprintf(stdout, "Lux = %d\n", sensor.getLux());
    }

    //! [Interesting]

    return (0);
}

File: examples/javascript/tsl2561.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2014 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

var digitalLightSensor = require('jsupm_tsl2561');

// Instantiate a digital light sensor TSL2561 on I2C
var myDigitalLightSensor = new digitalLightSensor.TSL2561();


setInterval(function()
{
	console.log("Light value is " + myDigitalLightSensor.getLux());
}, 1000);

// Print message when exiting
process.on('SIGINT', function()
{
	console.log("Exiting...");
	process.exit(0);
});
